### ISSUE TITLE
transfer learning for classification

### ISSUE THREAD

**Issue by FAFACHR** [dmlc/gluon-cv#835]
Hello @zhreshold, 
As you told me in this issue: #746 to do the fine tuning of 'resnet50_v1b' I should replace the 'finetune_net.output' with 'finetune_net.fc' and it works. But for some classifiers I need to use 'finetune_net.output', so can you explain what's the difference between the 'output' and 'fc' and why some classifiers need 'output' and the others need 'fc'.
Thank you in advance.

**Comment by zhreshold** [comment]
It's simply due to the name of the last layer written in network definition. We are trying to make all names as consistent as possible, however, due to the complexity of many new networks, it is sometimes not applicable.

**Comment by zhreshold** [comment]
You can always print the network ```print(net)``` to see the names of each layer

**Comment by FAFACHR** [comment]
Thank you @zhreshold, I have another question concerning the classifier's fine tuning. In the script I define the number of classes that I need to fine tune. Need to mention that I have a prepared dataset of k classes, however, if I give it a number of classes lesser or greater than k in the script, it actually proceeds to the fine tuning. I just don't get how this works. For example, if I give the script the number of classes "4", it takes the first 4 classes folders in the dataset?. How is the script reading the folders inside the dataset?

**Comment by zhreshold** [comment]
If you define classes=10  in network and only have 4 cat images in folder, it's totally fine. The network will never learn from samples belong to classes(5-10), and will most likely never predict class id larger than 4.

If you define classes=4 and prepared 10 cat images in folder, you might be in trouble now. The loss function is undefined given that the id is larger than number of outputs in network.

**Comment by FAFACHR** [comment]
Thank you @zhreshold, so as I understood if my dataset contain 4 folders so I have 4 classes : **folder_1, folder_2, folder_3, folder_4**, the script will assign a class id for each folder in a sorted way. So it will start with the first folder, so we will have:  **folder_1 : class_id= 0, folder_2 : class_id= 1,
folder_3 : class_id= 2, folder_4 : class_id= 3** but if I have folder_2 before folder_1 for some reason of sorting like : **folder_2, folder_1, folder_3, folder_4** we will have :  **folder_2 : class_id= 0, folder_1 : class_id= 1, folder_3 : class_id= 2, folder_4 : class_id= 3**
right?

**Comment by FAFACHR** [comment]
Please @zhreshold if you can reply to my comment to know how the script assign each class id for a folder class and thank you in advance.

**Comment by zhreshold** [comment]
the order should be alphabetic ascending order but your assumption looks correct

**Comment by FAFACHR** [comment]
Thank you for your help @zhreshold
